### PR TITLE
Merge Sort in Rust

### LINKED ISSUE #2136: Merge Sort
- [x] C
- [x] C++
- [x] C#
- [x] CoffeeScript
- [x] Dart
- [x] Go
- [x] Java
- [x] JS
- [ ] Kotlin - @shubhammantri1
- [x] Php
- [x] Python
- [x] Readme
- [x] Ruby
- [x] Rust - @m-code12
- [x] TypeScript - @m-code12

### PR DESCRIPTION
**Pull request by m-code12**
Fixes #2136 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Merge Sort in Rust 

#### Languages Used:
- Rust

#### Files Added:
- Merge_Sort.rs

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by m-code12** [comment]
I have implemented Merge Sort in Rust following all the required coding conventions. Please review and merge @jainaman224 @singh-shreya6

File: Merge_Sort/Merge_Sort.rs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+fn merge_split(a: &mut Vec<i32>, start: usize, end: usize, buffer: &mut Vec<i32>) {
+    if end - start > 1 {
+        // Determine the split index
+        let split = (start + end) / 2;
+		
+		// Split and merge sort
+        merge_split(a, start, split, buffer);
+        merge_split(a, split, end, buffer);
+        merge(a, buffer, start, split, end);
+       merge_copy(buffer, a, start, end);
```
Comment (singh-shreya6) [inline Merge_Sort/Merge_Sort.rs:26]: Indentation is not consistent here. Please ensure 4 space indentation inside the if block.

File: Merge_Sort/Merge_Sort.rs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    s.trim()
+        .split_whitespace()
+        .map(|word| word.parse())
+        .collect()
+}
+
+// Output :
+// Enter the numbers to be sorted (separated by space) : 
+// 76 56 45 3 1 4
+// Sorted array: [1, 3, 4, 45, 56, 76]
```
Comment (singh-shreya6) [inline Merge_Sort/Merge_Sort.rs:65]: Please add a blank line at the end of file

File: Merge_Sort/Merge_Sort.rs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    io::stdin()
+        .read_line(&mut s)
+        .expect("could not read from stdin");
+    s.trim()
+        .split_whitespace()
+        .map(|word| word.parse())
+        .collect()
+}
+
+// Output :
```
Comment (singh-shreya6) [inline Merge_Sort/Merge_Sort.rs:62]: Please convert this single line comments to one multi line comment.

File: Merge_Sort/Merge_Sort.rs [hunk 4]
```diff
@@ -0,0 +1,65 @@
+use std::io;
```
Comment (singh-shreya6) [inline Merge_Sort/Merge_Sort.rs:1]: Please add a brief problem description.

**Comment by m-code12** [comment]
I have made the respective changes . Please review @singh-shreya6 .

File: Merge_Sort/Merge_Sort.rs [hunk 5]
```diff
@@ -0,0 +1,102 @@
+/*
+ 
+            Rust Program to implement Merge Sort  
+     --------------------------------------------------------
+            
+                    Merge Sort 
+     --------------------------------------------------------
+            
+Merge sort is a divide-conquer-combine algorithm based on the idea of breaking down a list into several sub-lists until each sublist consists of a single element and merging those sublists in a manner that results into a sorted list.
```
Comment (singh-shreya6) [inline Merge_Sort/Merge_Sort.rs:9]: Please divide this line into 2 lines.

File: Merge_Sort/Merge_Sort.rs [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    -> Take adjacent pairs of two singleton lists and merge them to form a list of 2 elements.N will now convert into N/2 lists of size 2.
+    -> Repeat the process till a single sorted list of size N is obtained.
+
+While comparing two sublists for merging, the first element of both lists is taken into consideration.
+While sorting in ascending order, the element that is of a lesser value becomes a new element of the sorted list.
+This procedure is repeated until both the smaller sublists are empty and the new combined sublist comprises all the elements of both the sublists.
+
+Complexity Analysis :
+
+The list of size N is divided into a max of logN parts, and the merging of all sublists into a single list takes O(N) time, the worst case run time of this algorithm is O(NlogN) .
```
Comment (singh-shreya6) [inline Merge_Sort/Merge_Sort.rs:22]: Split this line also into 2 lines.

**Comment by singh-shreya6** [comment]
After making the requested changes. Please squash your commits also. So that it can be merged.

**Comment by m-code12** [comment]
I have made the final changes and also squashed my commits .

**Pull request merged**